### PR TITLE
gh-1870: Refactor `log_prob` method in `_MixtureBase` class to handle `-jnp.inf`

### LINKED ISSUE #1870: Grads w.r.t. weights of `MixtureGeneral` Distribution are giving `nan`s
Hi,

We have created some models where we estimate the weights of the `MixtureGeneral` distribution. However, when computing the gradient of this argument, we are encountering `nan` values. We enabled `jax.config.update("debug_nan", True)` to diagnose the issue, and it pointed to the following line:

https://github.com/pyro-ppl/numpyro/blob/8e9313fd64a34162bc1c08b20ed310373e82e347/numpyro/distributions/mixtures.py#L152

I suspect that after the implementation of https://github.com/pyro-ppl/numpyro/pull/1791, extra care is needed to handle `inf` and `nan` values, possibly by using a double `where` for a safe `logsumexp`.

> [!IMPORTANT]
> This is an urgent issue, so a prompt response would be greatly appreciated.

### PR DESCRIPTION
**Pull request by Qazalbash**
closes #1870

### CODE REVIEW

File: numpyro/distributions/mixtures.py [hunk 1]
```diff
@@ -149,7 +149,10 @@ def sample(self, key, sample_shape=()):
     def log_prob(self, value, intermediates=None):
         del intermediates
         sum_log_probs = self.component_log_probs(value)
-        return jax.nn.logsumexp(sum_log_probs, axis=-1)
+        safe_sum_log_probs = jnp.where(
+            jnp.isneginf(sum_log_probs), -jnp.inf, sum_log_probs
+        )
```
Comment (fehiepsi) [inline numpyro/distributions/mixtures.py:154]: Sounds like a grad bug of logsumexp to me.

**Review by fehiepsi — APPROVED** [review]

**Comment by fehiepsi** [comment]
Thanks @Qazalbash!
